### PR TITLE
Add readonly length properties to part helix and spiral

### LINKED ISSUE #5995: [Feature Request] Part: Show spiral length in the property editor (of Part Helix and Part Spiral (derived from Helix) objects)
Issue imported from https://tracker.freecad.org/view.php?id=4159

* **Reporter:** vocx
* **Date submitted:** 10/9/2019
* **FreeCAD version:** 
* **Category:** Feature
* **Status:** new
* **Tags:** 

# Original report text

Part Helix and Part Spiral (which are derived from Helix) objects don't show a length property in the property editor.

The length is included in the Shape, so it just has to be displayed as a read-only value.
```
obj.Shape.Length
```

See this thread [Length of a spiral](https://forum.freecadweb.org/viewtopic.php?f=3&t=39884)

# Other bug information

* **Priority:** normal
* **Severity:** feature
* **Category:** Feature
* **Updated:** 2/6/2021

### PR DESCRIPTION
**Pull request by bgbsww**
Fix #5995

### CODE REVIEW

File: src/Mod/Part/App/PrimitiveFeature.cpp [hunk 1]
```diff
@@ -799,6 +802,9 @@ App::DocumentObjectExecReturn *Helix::execute()
         Standard_Real myRadiusTop = myRadius + myHeight * tan(myAngle/180.0f*M_PI);
 
         this->Shape.setValue(TopoShape().makeSpiralHelix(myRadius, myRadiusTop, myHeight, nbTurns, mySegLen, myLocalCS));
+        GProp_GProps props;
+        BRepGProp::LinearProperties(Shape.getShape().getShape(), props);
+        Length.setValue(props.Mass());
```
Comment (yorikvanhavre) [inline src/Mod/Part/App/PrimitiveFeature.cpp:807]: Are you sure this is correct? Do we get the Mass to get the Length?

Comment (FlachyJoe) [reply src/Mod/Part/App/PrimitiveFeature.cpp:807]: I think it's ok.
[BRepGProp::LinearProperties Reference](https://dev.opencascade.org/doc/refman/html/class_b_rep_g_prop.html#ab1d4bacc290bfaa8df13dd99ae7b8e70):
> For this computation no linear density is attached to the edges. So, for example, the added mass corresponds to the sum of the lengths of the edges of S.

Comment (bgbsww) [reply src/Mod/Part/App/PrimitiveFeature.cpp:807]: That's consistent with how I researched this, and I couldn't find another way to get the Length.  I agree, it was weird.

Comment (kadet1090) [reply src/Mod/Part/App/PrimitiveFeature.cpp:807]: This could use comment in the code with reference to docs to not confuse other developers.

Comment (bgbsww) [reply src/Mod/Part/App/PrimitiveFeature.cpp:810]: That's a good idea.  Done.

**Pull request merged**